File: 07_programming/src/C/decrypt.c

void decrypt(const char* input, char* output) {
    char* pi;
    char* po;

    for (pi = (char*)input, po = output; *pi; pi++, po++) {
        if (*pi >= 'A' && *pi <= 'Z') {
            *po = 'A' + ((*pi - 'A' + 13) % 26);
        }
        else if (*pi >= 'a' && *pi <= 'z') {
            *po = 'a' + ((*pi - 'a' + 13) % 26);
        }
        else {
            *po = *pi;
        }

    }

    *(++po) = '\0';
}

File: 99_programming/src/Ruby/decrypt.rb
#!/usr/bin/env ruby

def decrypt(input)
    (input.each_char.map do |c|
        case c
          when ('a'..'z') then ('a'.ord + (c.ord - 'a'.ord + 13) % 26).chr
          when ('A'..'Z') then ('A'.ord + (c.ord - 'A'.ord + 13) % 26).chr
          else c
        end
    end).reduce(''){ |a, b| a + b }
end

puts decrypt('Fb, wrgmg unora Fvr nhpu rva ynhsraqrf Ehol-Cebtenzz.')
